src/scr.c

```diff
@@ -839,13 +839,7 @@ static int scr_get_params()
 
   /* set MPI buffer size (file chunk size) */
   if ((value = scr_param_get("SCR_MPI_BUF_SIZE")) != NULL) {
-    if (scr_abtoull(value, &ull) == SCR_SUCCESS) {
```

Comment:
@rhaas80 , you can see some examples of the strings this function can accept as input in the comments before the function: https://github.com/LLNL/scr/blob/38a5765e23dff13b6076410172fc2c84572c820a/src/scr_util.c#L83

Reply:
Thank you. I will add commits to use `scr_abtoull` the the places where numbers of bytes are used in parameters, where variables that are `int` are being an additional check for `<= INT_MAX` will be added (it seems too easy for someone to specify "3GB" otherwise).

Reply:
Right, good idea.

Reply:
I added code to use `scr_abtonull` to SCR in fdb1e95 and let this ripple through to the components which get byte like numbers using code similar to this:

```
if (kvtree_util_get_bytecount(config, ER_KEY_CONFIG_MPI_BUF_SIZE, &ul) ==
    KVTREE_SUCCESS) {
  er_mpi_buf_size = (int) ul;
  if (er_mpi_buf_size != ul) {
    fprintf(stderr, "Value %s passed for %s exceeds int range\n",
            value, ER_KEY_CONFIG_MPI_BUF_SIZE);
    retval = ER_FAILURE;
  }
}
```

ie they frist get them via `kvtree_util_get_bytecount` then cast and assign to their internal variable then check if an overflow happened. I am not checking against `INT_MAX` or to avoid issues with double where it is not double max that matters but the difference in mantissa size for a `double` (56bit or so I think) and the 64 bits in an `unsigned long`.

One could argue that the kvtree config interface targets code so that it should require actual numbers being passed rather than unit-full byte quantities, but then this adds same extra checks on the passed in value and requires less trust in the component's caller being sane.

Note that `kvtree_util_get_bytecount` and `scr_abtoull` are inconsistent in that the former sets an `unsigned long` while the latter sets `unsigned long long`. Should make a difference in practise unless one expects buffers sizes exceeding the 64bits in an `unsigned long` and requiring the 128 bits that `unsigned long long` typically provides.

Technically an `unsigned long` is required to be only 32bit, but the only system where i have recently found that to be true is (64bit) Windows so if one ever runs on Windows then `kvtree_util_get_bytecount` is not going to allow for byte counts >4GB.